plugins/builds/triggers/helpers.js

```diff
@@ -1032,6 +1033,48 @@ function buildsToRestartFilter(joinPipeline, groupEventBuilds, currentEvent, cur
         .filter(build => build !== null);
 }
 
+/**
+ * Get subsequent job names which the root is the start from node
+ * @param   {Array}   [workflowGraph]         Array of graph vertices
+ * @param   {Array}   [workflowGraph.nodes]   Array of graph vertices
+ * @param   {Array}   [workflowGraph.edges]   Array of graph edges
+ * @param   {String}  [startNode]             Starting/trigger node
+ * @returns {Array<String>}                   subsequent job names
+ */
+function subsequentJobFilter(workflowGraph, startNode) {
+    const { nodes, edges } = workflowGraph;
+
+    if (!startNode || !nodes.length) {
+        return [];
+    }
+
+    let start = startNode;
+
+    // startNode can be a PR job in PR events, so trim PR prefix from node name
+    if (startNode.match(/^PR-[0-9]+:/)) {
+        start = startNode.split(':')[1];
+    }
+
+    const visiting = [start];
+
+    const visited = new Set(visiting);
+
+    if (edges.length) {
+        while (visiting.length) {
+            const cur = visiting.shift();
```

Comment:
`Array.shift` is slower than `Array.pop`.
So using `visiting.pop()` would be better if we can use `pop` method here.

Reply:
I fixed it!